pageserver/src/aux_file.rs

```diff
@@ -0,0 +1,121 @@
+use pageserver_api::key::{Key, AUX_KEY_PREFIX, METADATA_KEY_SIZE};
+use tracing::warn;
+
+/// Create a metadata key from a hash, encoded as [AUX_KEY_PREFIX, 2B directory prefix, first 13B of 128b xxhash].
+fn aux_hash_to_metadata_key(
+    dir_level1: u8,
+    dir_level2: u8,
+    data: &[u8],
+) -> [u8; METADATA_KEY_SIZE] {
```

Comment:
This should return Key here directly.

Reply:
fixed in https://github.com/neondatabase/neon/pull/7401/commits/a11a333ceadeb86b130624b51021e10516c5d108